util/README_mtt_update.md

```diff
@@ -0,0 +1,151 @@
+
+## Preparing your source code
+
+This script makes assumptions about your source code. Before it is usable, you first have to prepare your source code accordingly.
+
+### Choosing the textdomain name
+
+It is recommended to set the textdomain name (for `minetest.get_translator`) to be identical of the mod name as the script will automatically detect it. If the textdomain name differs, you may have to manually change the `# textdomain:` line of newly generated files.
+
+**Note:** In each `*.tr` file, there **must** be only one textdomain. Multiple textdomains in the same file are not supported and any additional textdomain line will be removed.
+
+### Defining the helper functions
+
+In any source code file with translatable strings, you have to manually define helper functions at the top with something like `local S = minetest.get_translator("<textdomain>")`. Optionally, you can also define additional helper functions `FS`, `NS` and `NFS` if needed.
+
+Here is the list of all recognized function names. All functions return a string.
+
+* `S`: Returns translation of input. See Minetest's `lua_api.txt`. You should always have at least this function defined.
+* `NS`: Returns the input. Useful to make a string visible to the script without actually translating it here.
+* `FS`: Same as `S`, but returns a formspec-escaped version of the translation of the input. Supported for convenience.
+* `NFS`: Returns a formspec-escaped version of the input, but not translated. Supported for convenience.
+
+Here is the boilerplate code you have to add at the top of your source code file:
+
+    local S = minetest.get_translator("<textdomain>")
+    local NS = function(s) return s end
+    local FS = function(...) return minetest.formspec_escape(S(...)) end
+    local NFS = function(s) return minetest.formspec_escape(s) end
+
+Replace `<textdomain>` above and optionally delete `NS`, `FS` and/or `NFS` if you don't need them.
+
+### Preparing the strings
+
+Enclose any translatable strings in a function of the form `S("Some string")`. Only literal strings will be detected; variables will be ignored. For a string to be detected, it has to be written in a specific way that satisfies the following rules:
+
+* The function brackets must contain a single literal string, or a concatenation of literal strings
+* The function brackets **must not** be omitted
+* The string **must** use `"..."`, `'...'` or `[[...]]` as string delimiters
+* If it's a concatenation of literal strings, you can't use `[[...]]` as string delimiters
+* Concatenating a literal string with a variable is **not** permitted and will lead to errors
```

Comment:
A few examples might help to better understand these rules. What do you think about the following?

```suggestion
This script can detect translatable strings of the notations listed below. Additional
function arguments followed after a literal string are ignored.

* `S("literal")`: one literal string enclosed by the delimiters `"..."`, `'...'` or `[[...]]`
* `S("foo " .. 'bar ' .. "baz")`: concatenation of multiple literal strings. Line breaks are accepted.

Undetectable notations:

* `S"literal"`: omitted function brackets
* `S(variable)`: requires the use of `NS`. See example below.
* `S("literal " .. variable)`: non-static content. Use placeholders (`@1` ...) for dynamic translations.
* Any literal string concatenation using `[[...]]`
```

It might also be a good chance to put a note into lua_api.md` that `string.format` *must not be used* with translated strings. The server modifies the string such that the client cannot translate it any more.